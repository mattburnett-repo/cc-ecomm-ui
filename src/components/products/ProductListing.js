export default function ProductListing () {
    return (
        <div>
            <div role="presentation" aria-label="product-category">Product category</div>
            <div role="presentation" aria-label="product-categories">select product categories</div>
            <input aria-label="search-terms" placeholder="Enter search terms"></input>
            <button aria-label="search">Search</button>
            <div role="presentation" aria-label="product-listing">
                <div>product 01</div>
                <div>product 02</div>
                <div>product 03</div>
            </div>
        </div>
    )
}