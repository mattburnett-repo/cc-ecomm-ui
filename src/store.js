import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;