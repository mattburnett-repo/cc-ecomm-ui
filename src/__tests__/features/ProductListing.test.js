import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';
import store from '../../store'

import ProductListing from '../../features/products/productListing';

describe('<ProductListing /> feature tests', () => {
    // beforeEach(() => render(<Provider store={store}><ProductListing /></ Provider>));
    beforeEach(() => render(<Provider store={store}><ProductListing /></ Provider>));

    it('displays correct nav bar elements', () => {
        screen.getByRole('button', { name: /home/i});
        screen.getByRole('button', { name: /show-cart/i});
        screen.getByRole('button', { name: /logout/i});
    })
    it('displays products', () => {
        let theVals = screen.getAllByRole('presentation', {name: /^product$/i});
        expect(theVals.length).toBe(4);

        expect(theVals[0]).toHaveTextContent('product_name_01');
        expect(theVals[1]).toHaveTextContent('product_name_02');
        expect(theVals[2]).toHaveTextContent('product_name_03');
        expect(theVals[3]).toHaveTextContent('product_name_04');
    })

    it('creates a snapshot', () => {
        const tree = renderer.create(<Provider store={store}><ProductListing /></ Provider>).toJSON();
        expect(tree).toMatchSnapshot();  
    })
}); // end feature

describe('<ProductListing /> unit / functional tests', () => {
    beforeEach(() => render(<Provider store={store}><ProductListing /></ Provider>));

    it('selects a product category', () => {
        fireEvent.change(screen.getByRole('presentation', {name: /product-categories/i}), { target: { value: 2 } })
        let options = screen.getByRole('presentation', {name: /product-categories/i});
        expect(options[0].selected).toBeFalsy();
        expect(options[1].selected).toBeFalsy();
        expect(options[2].selected).toBeFalsy();
        expect(options[3].selected).toBeTruthy();
    });
    it('accepts search text', () => {
        const theElement = screen.getByRole('textbox', {name: /search-terms/i});
        userEvent.type(theElement, 'here is some search text');
        
        expect(theElement).toHaveValue('here is some search text');
    });
    it('clicks the search button', () => {
        const theVal = screen.getByRole('button', {name: /search/i});
        fireEvent.click(theVal);
    });
}); // end unit / functional