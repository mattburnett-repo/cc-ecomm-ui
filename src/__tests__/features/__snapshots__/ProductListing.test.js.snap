// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListing /> feature tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="nav-bar"
    role="presentation"
  >
    <div
      aria-label="header-message"
      role="presentation"
    >
      productListing
       
    </div>
    <button
      aria-label="home"
    >
      Home
    </button>
    <button
      aria-label="show-cart"
    >
      Show Cart
    </button>
    <button
      aria-label="logout"
    >
      Log Out
    </button>
  </div>
  <div
    aria-label="browse-search"
    role="presentation"
  >
    <select
      aria-label="product-categories"
      role="presentation"
    >
      <option
        value={4}
      >
        Auto
      </option>
      <option
        value={3}
      >
        Health
      </option>
      <option
        value={1}
      >
        Pets
      </option>
      <option
        value={2}
      >
        Tech
      </option>
    </select>
    <input
      aria-label="search-terms"
      placeholder="Enter search terms"
    />
    <button
      aria-label="search-button"
    >
      Search
    </button>
  </div>
  <div
    aria-label="products"
    role="presentation"
  >
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_01
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_01
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_01"
          src="item.image_url"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        1.00
      </div>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_02
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_02
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_02"
          src="item.image_url"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        2.00
      </div>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_03
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_03
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_03"
          src="item.image_url"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        3.00
      </div>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_04
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_04
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_04"
          src="item.image_url"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        4.00
      </div>
    </div>
  </div>
</div>
`;
